src/global/theme.tsx

```diff
@@ -124,21 +118,7 @@ function ThemeProviderInner({
         className={target ? undefined : classNames(className, themeClasses)}
         {...restProps}
       >
-        {passToPopups ? (
```

Comment:
This is the key point. I have moved logic of handling popups from Theme to the Popup – closer to the consumer. This also does not require maintaining portal anymore